x/lockup/keeper/lock.go

```diff
@@ -163,6 +187,28 @@ func (k Keeper) lock(ctx sdk.Context, lock types.PeriodLock, tokensToLock sdk.Co
 	return nil
 }
 
+// lockNoSend behaves the same as lock, but does not send the coins to the lockup module account.
+func (k Keeper) lockNoSend(ctx sdk.Context, lock types.PeriodLock, tokensToLock sdk.Coins) error {
```

Comment:
I realized the best way was to make a common function that they both call. It wouldn't have been feasible to call lockNoSend directly from lock because we want to ensure the send happens before the hook is called https://github.com/osmosis-labs/osmosis/pull/5162/commits/749b82d857dd1149122ac2bc410d104a21da17fd

Reply:
The following commit undoes what was done above in favor of your suggestion of making lockNoSend the default behavior https://github.com/osmosis-labs/osmosis/pull/5162/commits/d783cd5bcb795694c0dbbdf339b08604af12a8fb